CSS Counters Module (#29669)
mdn/content: files/en-us/web/css/@counter-style/system/index.md

/* Combined values */
system: fixed 3;

> system: extends decimal;

Author: I added " unless you are overriding or extending the style type" to the @counter-style definition

> system: extends decimal;

Reviewer: Thanks

system: extends customCounterName;
```

## Values

This may take one of three forms:

- One of the keyword values `cyclic`, `numeric`, `alphabetic`, `symbolic`, `additive`, or `fixed`.
- The keyword value `fixed` along with an integer.
- The keyword value or `extends` along with `<counter-style>`, which is a [`<counter-style-name>`](/en-US/docs/Web/CSS/@counter-style#counter-style-name) or {{cssxref("symbols", "symbols()")}}.

> The values include:

Author: done.

- `cyclic`

  - : Cycles through the list of symbols provided in [the `symbols` descriptor](/en-US/docs/Web/CSS/@counter-style/symbols). Once the end of the list of symbols is reached, it will loop back to the beginning and start over. This system is useful for simple bullet styles with just one symbol, or for styles having multiple symbols. At least one symbol must be specified in the `symbols` descriptor, or the counter style is not valid.

> - `fixed` or `fixed <integer>`
> 
>   - : Defines a finite set of symbols are specified; iterating once through the list of symbols provided in the `symbols` descriptor. Once the system iterated through the symbols specified, the fallback counter style is used. This system is useful in cases where the counter values are finite. At least one symbol must be specified in the `symbols` descriptor or the counter style is not valid. The `fixed` value can include an optional {{cssxref("&lt;integer&gt;")}}. If specified, the value indicates which item will get the first symbol in the single iteration of the fixed list of symbols as its marker. By default, if the integer is omitted, the value of the first integer is taken as `1`.

Reviewer: Last two sentences are not clear. Does the integer mark an item from the `symbols` list or the item in the content list? I've made some edits based on my understanding - please check for technical accuracy.

```suggestion
- `fixed` or `fixed <integer>`

  - : Defines a finite set of symbols, iterating once through the list of symbols provided in the `symbols` descriptor. Once the specified symbols have been iterated through, the fallback counter style is used. This keyword value is useful in cases where the counter style values are finite. At least one symbol must be specified in the `symbols` descriptor, otherwise the counter style is not valid. The `fixed` keyword can be followed by an optional {{cssxref("&lt;integer&gt;")}} value. If specified, the `<integer>` value indicates the item in the list that will get the first symbol from the list of symbols. If omitted, the default value of `integer` is `1`, which gives the first item in the list the first symbol.
```

- `symbolic`

>   - : Cycles through the symbols provided in the `symbols` descriptor list of symbols repeatedly, doubling, tripling, etc. the symbols on each successive pass through the list. For example, if there were are two symbols provided for the `symbols` descriptor, "◽" and "◾", on each successive pass, they will become "◽◽" and "◾◾", "◽◽◽" and "◾◾◾" and so on. At least one symbol must be specified in the `symbols` descriptor or the counter style is not valid. This counter system works for positive counter values only.

Reviewer: ```suggestion
  - : Cycles through the symbols provided in the `symbols` descriptor list repeatedly, doubling, tripling, and so on, the symbols on each successive pass through the list. For example, if two symbols "◽" and "◾" are specified in the `symbols` descriptor, on each successive pass, they will become "◽◽" and "◾◾" and "◽◽◽" and "◾◾◾" and so on in subsequent passes. At least one symbol must be specified in the `symbols` descriptor, otherwise the counter style is not valid. This counter system works for positive counter values only.
```

- `extends`

>   - : Allows authors to use extend the algorithm of another browser or author defined counter style by altering its other aspects. Any unspecified descriptors, and their values will be taken from the extended counter style specified. If the specified counter style name in extends, is not a currently defined counter style name, it will instead extend from the decimal counter style.

Reviewer: ```suggestion
  - : Extends the algorithm of another browser- or author-defined counter style by allowing the alteration of some aspects of the extended counter style. Any unspecified descriptors and their values are inherited from the extended counter style specified. If the counter style name specified with `extends` is not yet defined, the `decimal` counter style will be extended by default.
```

>     It must not contain a `symbols` or `additive-symbols` descriptor, or else the counter style rule is invalid. If one or more counter styles definitions form a cycle with their extends values, the browser will treat all the participating counter styles as extending from the decimal style.

Reviewer: ```suggestion
    It must not contain a `symbols` or `additive-symbols` descriptor, otherwise the counter style rule will be invalid. If one or more counter styles definitions form a cycle with their `extends` values, the browser will treat all the participating counter styles as extending from the `decimal` style.
```

### Fixed counter

> The `fixed` value iterates through the list of symbols only once, starting the single cycle at the number indicated by the integer:

Reviewer: ```suggestion
The `fixed` value iterates through the list of symbols only once, starting the single cycle at the list item number indicated by the `integer` value:
```

### Symbolic counter

> The `symbolic` value loops throught the list defined in the `symbols` descriptor, doubling and tripling the number of symbols for the second and third cycles through the list, respectively:

Reviewer: ```suggestion
The `symbolic` value loops through the list defined in the `symbols` descriptor, doubling and tripling the number of symbols for the second and third cycles through the list, respectively:
```

### Numeric counter

The first symbol provided in the `symbols` descriptor is interpreted as `0` here.

> #### CSS<br>

Reviewer: ```suggestion
#### CSS
```

#### HTML

> We included the [`start`](/en-US/docs/Web/HTML/Element/ol#start) attribute on the {{HTMLElement("ol")}} to demonstrate that counting doesn't need to start with `1` and the [`value`](/en-US/docs/Web/HTML/Element/li#value) attribute on the fifth {{HTMLElement("li")}} element to demonstrate that counters you define using `@counter-style` behave just like native counters.

Reviewer: ```suggestion
We use the [`start`](/en-US/docs/Web/HTML/Element/ol#start) attribute on the {{HTMLElement("ol")}} element to demonstrate that counting doesn't need to start at `1`. Additionally, we use the [`value`](/en-US/docs/Web/HTML/Element/li#value) attribute on the fifth {{HTMLElement("li")}} element to demonstrate that the counters you define using `@counter-style` behave just like native counters.
```

> ### Extends example

Reviewer: ```suggestion
### Extending a counter
```

> This example will use the algorithm, symbols, and other properties of [`lower-alpha`](/en-US/docs/Web/CSS/list-style-type#lower-alpha), one of the several native {{CSSXref("list-style-type")}} counter values, but extends it, removing the period (`'.'`) after the counter representation, and enclosing the characters in parenthesis; like `(a)`, `(b)`, etc.

Reviewer: ```suggestion
This example uses the algorithm, symbols, and other properties of [`lower-alpha`](/en-US/docs/Web/CSS/list-style-type#lower-alpha), one of the several native {{CSSXref("list-style-type")}} counter values, but extends it by removing the period (`'.'`) after the counter representation and enclosing the characters in parentheses, as in  `(a)` and `(b)`.
```
